### DISCUSSION TITLE
运行 ConsoleInject.exe  -m 出现 Unknown option character `\x6d'.

### DISCUSSION THREAD

**Discussion by weihua-git** [ttttupup/wxhelper#116] [Q&A]
大佬帮忙看看，多开好像不行
![image](https://user-images.githubusercontent.com/59214326/234161933-6f67d51e-68d8-4698-99fe-bac6f8f382d3.png)

**Comment by ttttupup** [comment]
add_subdirectory(source) 把cmake里的这个注释去掉，重新编译一个，这个编译好的，我也记不清哪个是最新的了。

**Comment by ttttupup** [comment]
release里附带了一个编译好的注入工具

**Comment by weihua-git** [comment]
好的，看到了，感谢

**Comment by weihua-git** [comment]
![image](https://user-images.githubusercontent.com/59214326/234212717-d8d24957-34b4-49d4-813f-65c51b8a043d.png)
现在用ConsoleApplication.exe 可以实现-m    ConsoleInject.exe 不可以，但是 ConsoleApplication.exe不能实现-i 注入那些，注入结果一直是0
![image](https://user-images.githubusercontent.com/59214326/234213090-a021672d-c81b-4cde-8163-269960ecc3c5.png)
ConsoleInject.exe  可以注入除了-m，但是2个都不能带端口 
ConsoleInject.exe 出现
![image](https://user-images.githubusercontent.com/59214326/234213282-20e83129-46f0-4462-877f-2e01b5bfbf9c.png)
 ConsoleApplication.exe 出现
![image](https://user-images.githubusercontent.com/59214326/234213379-7a21d931-68b3-407a-9f79-7cbfda558f07.png)

**Comment by weihua-git** [comment]
![image](https://user-images.githubusercontent.com/59214326/234215314-3b229629-eedf-460d-bb75-514395f9da1d.png)

**Comment by ttttupup** [comment]
[3.9.0.28-specify-port-V13](https://github.com/ttttupup/wxhelper/releases/tag/3.9.0.28-specify-port-V13) 这个版本的才是指定端口的版本

**Comment by weihua-git** [comment]
试了下 3.9.0.28 1.必须装指定版本的微信，新版会报错，2.第一次注入的端口能访问，后面注入的都访问不了，就算登了2个微信，先-m了一个微信也不行 ，
可否实现多开多个微信 每个注入指定一个端口
![image](https://user-images.githubusercontent.com/59214326/234444505-73551edd-522f-48bc-bf1a-e76511a141c6.png)
![image](https://user-images.githubusercontent.com/59214326/234444573-e996d74c-9c45-4e43-9f15-78f6ca4de5f2.png)

![image](https://user-images.githubusercontent.com/59214326/234444518-411b1f0b-e60f-43a3-b656-be65e156a188.png)
![image](https://user-images.githubusercontent.com/59214326/234444553-ccd0c3ad-ab63-46e0-b6d4-407761e33556.png)

**Comment by ttttupup** [comment]
这个就是多开的，-m参数 传入已经启动的微信的pid，这个就是删_WeChat_App_Instance_Identity_Mutex_Name这个句柄的。删完后才能再启动下一个，直接换端口就可以了。

**Comment by ttttupup** [comment]
这个注入工具的问题，是根据应用名弄的，不是根据pid的，所以第二次才会注入失败。

**Comment by weihua-git** [comment]
我后面试了一个先-m掉一个登录了2个微信，然后换端口注入还是访问不了就是最后一张图

**Comment by ttttupup** [comment]
注入工具那里改成pid就行了，这个注入工具就是随便写的用来测试的。第二次注入应该注入的还是第一个

**Comment by weihua-git** [comment]
就是要改源码重新编译一个吗，我这边不太懂这个源码，编译都不太会🤦‍

**Comment by ttttupup** [comment]
后面优化一下这个工具

**Comment by ttttupup** [comment]
> 就是要改源码重新编译一个吗，我这边不太懂这个源码，编译都不太会🤦‍

3.9.2.23-V5 版本release里更新了这个注入工具。用这个即可

**Comment by weihua-git** [comment]
我刚试了下大佬，但是好像注入不了，
![image](https://user-images.githubusercontent.com/59214326/234461287-ffaf1e34-0b84-4c7c-8df5-61b65eb9260a.png)
![1682479256209](https://user-images.githubusercontent.com/59214326/234461216-4731edbe-4a33-4f95-8ce3-9606d019e0c1.png)

**Comment by weihua-git** [comment]
![image](https://user-images.githubusercontent.com/59214326/234461832-a344af04-d679-464c-b989-ceb3c83d95a2.png)

**Comment by ttttupup** [comment]
[3.9.0.28-specify-port-V13](https://github.com/ttttupup/wxhelper/releases/tag/3.9.0.28-specify-port-V13) 这个版本的才是指定端口的版本 ,微信版本也要是3.9.0.28。

**Comment by ttttupup** [comment]
先用3.9.2.23-V5 的版本不指定端口的试试能否注入成功

**Comment by ttttupup** [comment]
https://github.com/ttttupup/wxhelper/discussions/65

**Comment by weihua-git** [comment]
V5 版本不指定端口可以注入成功

**Comment by ttttupup** [comment]
用v6 的 配置文件里改端口
